packages/api/src/router/widgets/health-monitoring.ts

```diff
@@ -0,0 +1,60 @@
+import { TRPCError } from "@trpc/server";
+import { observable } from "@trpc/server/observable";
+
+import type { HealthMonitoring } from "@homarr/integrations";
+import { integrationCreator } from "@homarr/integrations";
+import { logger } from "@homarr/log";
+import { createItemAndIntegrationChannel } from "@homarr/redis";
+
+import { createManyIntegrationMiddleware } from "../../middlewares/integration";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+export const healthMonitoringRouter = createTRPCRouter({
+  getHealthStatus: publicProcedure
+    .unstable_concat(createManyIntegrationMiddleware("query", "openmediavault"))
+    .query(async ({ ctx }) => {
+      const results = await Promise.all(
+        ctx.integrations.map(async (integration) => {
+          const client = integrationCreator(integration);
+
+          try {
+            const healthInfo = await client.getSystemInfoAsync();
```

Comment:
Do not call functions, that call integration APIs, in the TPC procedures.
We no longer do this because...
- this will not scale well with many users. Imagine that every client is invoking this procedure every 5 seconds. You have 100 users with 2 tabs of Homarr opened a the same time. You'll get approx. 1'000 API requests per minute which you will forward to the integration. Many integrations will break or refuse this many requests
- the ``await`` keyword will let the Promise resolve. This is bad because the procedure will have to wait for it to resolve. If the API request to the integration takes 8 seconds due to slow network, the procedure will be fetching for 8 seconds. This can lead to lags and suboptimal performance
- We want to pre-fetch data so data is already available before the user loads the page.
- We want to handle fetching asynchronously and handle errors in a more streamlined approach

As a summary; please move this call to the job and get the latest value from the in-memory Redis DB.